sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -575,12 +583,21 @@ public DataSourceConfiguration withMaxConnections(
     }
 
     /**
-     * Sets the class loader instance to be used to load the JDBC driver. If not specified, the
-     * default class loader is used.
+     * Sets the directory to the JDBC driver in the worker. If specified, the path is used to load
+     * the driver jar. If not specified, the default class loader is used to load the jars.
+     *
+     * <p>NOTE - driverJarDirectory must be the absolute path to the directory.
      */
-    public DataSourceConfiguration withDriverClassLoader(ClassLoader driverClassLoader) {
-      checkArgument(driverClassLoader != null, "driverClassLoader can not be null");
-      return builder().setDriverClassLoader(driverClassLoader).build();
+    public DataSourceConfiguration withDriverJarDirectory(String driverJarDirectory) {
+      checkArgument(driverJarDirectory != null, "driverJarDirectory can not be null");
+      return withDriverJarDirectory(ValueProvider.StaticValueProvider.of(driverJarDirectory));
+    }
+
+    /** Same as {@link #withDriverJarDirectory(String)} but accepting a ValueProvider. */
+    public DataSourceConfiguration withDriverJarDirectory(
+        ValueProvider<@Nullable String> driverJarDirectory) {
```

Comment:
I believe, ValueProviders are not needed with Flex templates, right ?

Reply:
I added value providers since we want to migrate classic templates using DynamicJdbcIO to JdbcIO as well.
ValueProviders are needed for classic templates